python/desc/imsim/imSim.py

```diff
@@ -343,58 +350,77 @@ def sources_from_list(object_lines, obs_md, phot_params, file_name,
         message += "    %d had n_points <= 0 \n" % n_bad_knots
         warnings.warn(message)
 
-    wav_int = None
-    wav_gal = None
+    target_chips = [target_chip] if target_chip is not None else None
```

Comment:
From the doctoring etc  if target_chip is None then you should use all sensors. This is accomplished here by having the target_chips=None interpreted in `_chip_downselect` as meaning you should use all of them with `[det.getName() for det in lsst_camera()]`.

FWIW I found this confusing when I first read the code.  This is far enough done in the function that a comment here and in the docstring of `_chip_downselect` would probably be useful.